File: src/test/java/javabot/dao/ChannelDaoTest.java
package javabot.dao;

import java.util.List;

import javabot.model.Channel;
import org.springframework.beans.factory.annotation.Autowired;
import org.testng.Assert;
import org.testng.annotations.Test;

public class ChannelDaoTest extends BaseServiceTest {
    @Autowired
    private ChannelDao channelDao;

    @Test
    public void addChannel() {
        final String testing = getTestBot().getNick();
        final Channel channel = new Channel();
        channel.setName(testing);
        channel.setLogged(true);
        channelDao.save(channel);
        Assert.assertNotNull(channelDao.get(testing));
    }

    @Test
    public void getChannels() {
        final List channels = channelDao.getChannels();
        Assert.assertTrue(!channels.isEmpty());

    }

    @Test
    public void currentChannels() {
        final List channels = channelDao.configuredChannels();
        Assert.assertTrue(!channels.isEmpty());
    }
}

File: src/test/kotlin/javabot/dao/ChannelDaoTest.kt
package javabot.dao

import com.mongodb.BasicDBObject
import dev.morphia.DeleteOptions
import javabot.model.Channel
import org.testng.Assert
import org.testng.annotations.Test

@Test
class ChannelDaoTest : BaseServiceTest() {
    fun addChannel() {
        val channel = Channel()
        val name = "##" + System.currentTimeMillis()
        channel.name = name
        channel.logged = true
        channelDao.save(channel)
        Assert.assertNotNull(channelDao.get(name))
    }

    fun getChannels() {
        Assert.assertTrue(!channelDao.getChannels().isEmpty())

    }

    fun currentChannels() {
        Assert.assertTrue(!channelDao.configuredChannels().isEmpty())
    }

    @Test(enabled = false)
    fun stats() {
        datastore.find(Channel::class.java)
                .delete(DeleteOptions().multi(true))
        val list = channelDao.getStatistics()
        Assert.assertTrue(list.isNotEmpty())
        val activity = list[0]
        Assert.assertNotSame(activity.total, 0)
        Assert.assertNotNull(activity.getPercent())
    }
}